core/dbt/task/base.py

```diff
@@ -77,6 +77,30 @@ def __init__(self, args, config, project=None):
         self.config = config
         self.project = config if isinstance(config, Project) else project
 
+        if dbt.tracking.active_user is not None:
+
+            # N.B. The none checking below is largely due to incomplete projects used in the testing
+            # and to support tasks that don't require a complete project or a complete config (debug, init).
```

Comment:
As discussed live with @iknox-fa: Let's consider moving these tracking calls to the `@requires` functions where we load (or are passed) the relevant bits of info, since those are called by both CLI + `dbtRunner` invocation mechanisms.
- `@requires.profile` for `adapter_type` + `adapter_unique_id`
- `@requires.project` for `project_id`

Reply:
Done!